mobile/src/main/java/rs/readahead/washington/mobile/views/activity/camera/CameraViewModel.kt

```diff
@@ -38,48 +41,83 @@ class CameraViewModel @Inject constructor() : ViewModel() {
     private val _lastMediaFileError = SingleLiveEvent<Throwable>()
     val lastMediaFileError: LiveData<Throwable> = _lastMediaFileError
 
+    private val _lastBackgroundActivityModel = SingleLiveEvent<BackgroundActivityModel>()
+    val lastBackgroundActivityModel: LiveData<BackgroundActivityModel> =
+        _lastBackgroundActivityModel
+
     fun addJpegPhoto(jpeg: ByteArray, parent: String?) {
-        disposables.add(
-            Observable.fromCallable { MediaFileHandler.saveJpegPhoto(jpeg, parent) }
-                .doOnSubscribe { _addingInProgress.postValue(true) }
-                .observeOn(AndroidSchedulers.mainThread())
-                .doFinally { _addingInProgress.postValue(false) }
-                .subscribe(
-                    { bundle -> _addSuccess.postValue(bundle.blockingGet()) },
-                    { throwable ->
-                        FirebaseCrashlytics.getInstance().recordException(throwable)
-                        _addError.postValue(throwable)
-                    }
+        disposables.add(Observable.fromCallable { MediaFileHandler.saveJpegPhoto(jpeg, parent) }
+            .doOnSubscribe {
+                val backgroundVideoFile = BackgroundActivityModel(
```

Comment:
We should rename backgroundVideoFile to something like backgroundFile since it could be photo or video and we are adding mimeType.